Site/dataobjects/SiteMedia.php

```diff
@@ -745,7 +754,7 @@ public function getFilename($encoding_shortname)
 		if ($this->getMediaSet()->obfuscate_filename) {
 			$filename = $this->filename;
 		} else {
-			$filename = $this->id;
+			$filename = $encoding_shortname;
```

Comment:
I think this change needs to be moved down to SiteVideoMedia and wrapped in a `if ($this->has_hls) {}` statement.